### DISCUSSION TITLE
Add model_id to DeviceInfo

### DISCUSSION THREAD

**Discussion by joostlek** [home-assistant/architecture#1104] [Entity Models] [answered]
Currently, the `DeviceInfo` provides a `model` for the integration to provide the model information of a device. A lot of devices have both a model name and an identifier. I think a phone is the best example to show an example of where this is used. A Samsung Galaxy S24 (which is the model name) has a series of model identifiers: `SM-S921B` (international version), `SM-S921B/DS` (international version with dual sim), `SM-S921U` (US version), `SM-S921U1` (Unlocked US version), (and 5 other version but you got the point now).

In the Home Assistant ecosystem this can also be a case. For example, Tuya has a lot of devices that have a certain name and a model identifier. I think that there are many more examples from within the core codebase where we have a good difference between a more human readable model name and a model identifier.

I want to propose to add a `model_id` property to `DeviceInfo`. This way we can keep track of these identifier without having them squished together into one field, to avoid a `Samsung Galaxy S24 (SM-S921B)` model. The current `model` field should then be utilized for the model name.

Apart from increasing the correctness of the fields, we also increase the data for a future device database. Since for users its easier to see that you support a `Samsung Galaxy S24` than a `Samsung Galaxy S24 (SM-S921B)`, `Samsung Galaxy S24 (SM-S921B/DS)` and `Samsung Galaxy S24 (SM-S921U)`.

Changelog:
- 15 June 2024 - Changed from `model_name` to `model_id`

**Comment by emontnemery** [comment]
Let's instead add model_name + model_number, and don't allow integrations to set model + one of the other two

**Reply by balloob** [reply]
We don't really care about model name as an identifier, and so it being polluted with model ID is fine (it would also be the thing shown if model name is not set). 

I am in favor of adding just `model_id`, and ensure it's a never changing model ID number with strict rules.

**Comment by frenck** [comment, marked as answer]
As discussed, this one if approved and ready to go 👍 

../Frenck

**Reply by joostlek** [reply]
https://github.com/home-assistant/core/pull/121989

**Reply by frenck** [reply]
Closing as it has been merged.
